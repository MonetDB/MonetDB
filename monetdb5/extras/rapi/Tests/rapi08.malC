gv:= bat.new(:int);
gv:= bat.append(gv,1);
gv:= bat.append(gv,1);
gv:= bat.append(gv,2);
gv:= bat.append(gv,2);

dv:= bat.new(:int);
dv:= bat.append(dv,42);
dv:= bat.append(dv,84);
dv:= bat.append(dv,42);
dv:= bat.append(dv,20);

# tuple identifiers
ti:= bat.new(:oid);
ti:= bat.append(ti,0:oid);
ti:= bat.append(ti,1:oid);
ti:= bat.append(ti,2:oid);
ti:= bat.append(ti,3:oid);

ti_gv:bat[:int] := algebra.projection(ti,gv);
(grp:bat[:oid],r1_9:bat[:oid],r2_9:bat[:lng]) := group.groupdone(ti_gv);
ti_dv:bat[:int] := algebra.projection(ti,dv); 

io.print(grp);
res:bat[:dbl] := rapi.subeval_aggr(nil:ptr,"res <- aggregate(arg2, by=list(aggr_group), FUN=median); return(res$x)",ti_gv,ti_dv);
io.print(res);
