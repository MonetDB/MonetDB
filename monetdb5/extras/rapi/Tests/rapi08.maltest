statement ok
gv:= bat.new(:int)

statement ok
gv:= bat.append(gv,1)

statement ok
gv:= bat.append(gv,1)

statement ok
gv:= bat.append(gv,2)

statement ok
gv:= bat.append(gv,2)

statement ok
dv:= bat.new(:int)

statement ok
dv:= bat.append(dv,42)

statement ok
dv:= bat.append(dv,84)

statement ok
dv:= bat.append(dv,42)

statement ok
dv:= bat.append(dv,20)

statement ok
ti:= bat.new(:oid)

statement ok
ti:= bat.append(ti,0:oid)

statement ok
ti:= bat.append(ti,1:oid)

statement ok
ti:= bat.append(ti,2:oid)

statement ok
ti:= bat.append(ti,3:oid)

statement ok
ti_gv:bat[:int] := algebra.projection(ti,gv)

statement ok
(grp:bat[:oid],r1_9:bat[:oid],r2_9:bat[:lng]) := group.groupdone(ti_gv)

statement ok
ti_dv:bat[:int] := algebra.projection(ti,dv)

query II rowsort
io.print(grp)
----
0
0
1
0
2
1
3
1

statement ok
res:bat[:dbl] := rapi.subeval_aggr(nil:ptr,"res <- aggregate(arg2, by=list(aggr_group), FUN=median); return(res$x)",ti_gv,ti_dv);

query IR nosort
io.print(res)
----
0
63.000
1
31.000

