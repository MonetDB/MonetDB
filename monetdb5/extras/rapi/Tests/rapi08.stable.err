stderr of test 'rapi08` in directory 'monetdb5/extras/rapi` itself:


# 20:20:45 >  
# 20:20:45 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=38117" "--set" "mapi_usock=/var/tmp/mtest-2827/.s.monetdb.38117" "--set" "monet_prompt=" "--forcemito" "--set" "mal_listing=2" "--dbpath=/ufs/manegold/_/Monet/HG/Oct2014/prefix/--disable-debug_--enable-assert_--enable-optimize/var/MonetDB/mTests_monetdb5_extras_rapi" "--set" "mal_listing=2" "--set" "embedded_r=yes"
# 20:20:45 >  

# builtin opt 	gdk_dbpath = /ufs/manegold/_/Monet/HG/Oct2014/prefix/--disable-debug_--enable-assert_--enable-optimize/var/monetdb5/dbfarm/demo
# builtin opt 	gdk_debug = 0
# builtin opt 	gdk_vmtrim = no
# builtin opt 	monet_prompt = >
# builtin opt 	monet_daemon = no
# builtin opt 	mapi_port = 50000
# builtin opt 	mapi_open = false
# builtin opt 	mapi_autosense = false
# builtin opt 	sql_optimizer = default_pipe
# builtin opt 	sql_debug = 0
# cmdline opt 	gdk_nr_threads = 0
# cmdline opt 	mapi_open = true
# cmdline opt 	mapi_port = 38117
# cmdline opt 	mapi_usock = /var/tmp/mtest-2827/.s.monetdb.38117
# cmdline opt 	monet_prompt = 
# cmdline opt 	mal_listing = 2
# cmdline opt 	gdk_dbpath = /ufs/manegold/_/Monet/HG/Oct2014/prefix/--disable-debug_--enable-assert_--enable-optimize/var/MonetDB/mTests_monetdb5_extras_rapi
# cmdline opt 	mal_listing = 2
# cmdline opt 	embedded_r = yes
# cmdline opt 	gdk_debug = 536870922

# 20:20:46 >  
# 20:20:46 >  "mclient" "-lmal" "-ftest" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-2827" "--port=38117"
# 20:20:46 >  


# 20:20:46 >  
# 20:20:46 >  "Done."
# 20:20:46 >  

