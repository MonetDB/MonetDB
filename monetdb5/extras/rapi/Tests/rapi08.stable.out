stdout of test 'rapi08` in directory 'monetdb5/extras/rapi` itself:


# 20:20:45 >  
# 20:20:45 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=38117" "--set" "mapi_usock=/var/tmp/mtest-2827/.s.monetdb.38117" "--set" "monet_prompt=" "--forcemito" "--set" "mal_listing=2" "--dbpath=/ufs/manegold/_/Monet/HG/Oct2014/prefix/--disable-debug_--enable-assert_--enable-optimize/var/MonetDB/mTests_monetdb5_extras_rapi" "--set" "mal_listing=2" "--set" "embedded_r=yes"
# 20:20:45 >  

# MonetDB 5 server v11.19.0
# This is an unreleased version
# Serving database 'mTests_monetdb5_extras_rapi', using 8 threads
# Compiled for x86_64-unknown-linux-gnu/64bit with 64bit OIDs dynamically linked
# Found 15.591 GiB available main-memory.
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008-2015 MonetDB B.V., all rights reserved
# Visit http://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://rome.ins.cwi.nl:38117/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-2827/.s.monetdb.38117
# MonetDB/GIS module loaded
# MonetDB/SQL module loaded
# MonetDB/R   module loaded


# 13:27:36 >  
# 13:27:36 >  "mclient" "-lmal" "-ftest" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-9249" "--port=37225"
# 13:27:36 >  

#io.print(X_9);
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	0@0	]
[ 1@0,	0@0	]
[ 2@0,	1@0	]
[ 3@0,	1@0	]
#io.print(X_16);
#--------------------------#
# h	t  # name
# void	dbl  # type
#--------------------------#
[ 0@0,	63	]
[ 1@0,	31	]

# 13:27:36 >  
# 13:27:36 >  "Done."
# 13:27:36 >  

